File: ASP Advanced/Arrays/Two Pointers/Leetcode-80.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int start = 0;
        int i = 0;
        while (i < nums.size()){
            nums[start++] = nums[i++];
            while (i > 0 && i < nums.size() - 1 && nums[i - 1] == nums[i + 1]){
                i++;
            }
        }
        return start;
    }
};

// The code defines a solution to the problem of removing duplicates from a sorted array.
// It uses a two-pointer technique where one pointer (`start`) keeps track of the position to insert unique elements,
// and the other pointer (`i`) iterates through the array. 
// The function returns the new length of the array after removing duplicates, while modifying the input array in place.

File: Python/ASP Advanced/Arrays/Two Pointers/Leetcode-80.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii/submissions/1533575298/

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = 0
        for elem in nums:
            if l == 0 or l == 1 or nums[l - 2] != elem:
                nums[l] = elem
                l += 1

        return l